pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/queries/QueryRunnerTestBase.java

```diff
@@ -193,167 +197,198 @@ protected List<Object[]> queryH2(String sql)
+        return value;
+      case INT_ARRAY:
+        if (value instanceof JdbcArray) {
+          try {
+            Object[] array = (Object[]) ((JdbcArray) value).getArray();
+            int[] intArray = new int[array.length];
+            for (int i = 0; i < array.length; i++) {
+              intArray[i] = ((Number) array[i]).intValue();
             }
+            return intArray;
+          } catch (SQLException e) {
+            throw new RuntimeException(e);
           }
-        } catch (SQLException e) {
-          throw new RuntimeException(e);
         }
-        return 0;
-      } else if (l instanceof String[]) {
-        String[] larray = (String[]) l;
-        try {
-          if (r instanceof JdbcArray) {
-            Object[] rarray = (Object[]) ((JdbcArray) r).getArray();
-            for (int idx = 0; idx < larray.length; idx++) {
-              if (!larray[idx].equals(rarray[idx])) {
-                return -1;
-              }
-            }
-          } else {
-            String[] rarray = (r instanceof List) ? ((List<String>) r).toArray(new String[0]) : (String[]) r;
-            for (int idx = 0; idx < larray.length; idx++) {
-              if (!larray[idx].equals(rarray[idx])) {
-                return -1;
-              }
-            }
-          }
-        } catch (SQLException e) {
-          throw new RuntimeException(e);
+        assertTrue(value instanceof int[],
+            "Got unexpected value type: " + value.getClass() + " for INT_ARRAY column, expected: int[] or JdbcArray");
+        return value;
+      case STRING_ARRAY:
+        if (value instanceof List) {
+          return ((List) value).toArray(new String[0]);
         }
-        return 0;
-      } else if (l instanceof JdbcArray) {
-        try {
-          Object[] larray = (Object[]) ((JdbcArray) l).getArray();
-          Object[] rarray = (Object[]) ((JdbcArray) r).getArray();
-          for (int idx = 0; idx < larray.length; idx++) {
-            if (!larray[idx].equals(rarray[idx])) {
-              return -1;
+        if (value instanceof JdbcArray) {
+          try {
+            Object[] array = (Object[]) ((JdbcArray) value).getArray();
+            String[] stringArray = new String[array.length];
+            for (int i = 0; i < array.length; i++) {
+              stringArray[i] = (String) array[i];
             }
+            return stringArray;
+          } catch (SQLException e) {
+            throw new RuntimeException(e);
           }
-        } catch (SQLException e) {
-          throw new RuntimeException(e);
         }
+        assertTrue(value instanceof String[], "Got unexpected value type: " + value.getClass()
+            + " for STRING_ARRAY column, expected: String[], List or JdbcArray");
+        return value;
+      default:
+        throw new UnsupportedOperationException("Unsupported ColumnDataType: " + columnDataType);
+    }
+  }
+
+  protected static void sortRows(List<Object[]> rows) {
```

Comment:
we might need to expose sortRows and typeCompatibleFuzzyCompare to integration test b/c some tests are failing b/c of the ordering doesn't match between H2 and Pinot